modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -368,9 +362,12 @@ private <R> CompletableFuture<R> enlistInTx(
         return postEnlist(fut, false, actualTx, actualTx.implicit()).handle((r, e) -> {
             if (e != null) {
                 if (actualTx.implicit()) {
+                    // TODO: IGNITE-24244
+                    long timeout = actualTx.isReadOnly() ? actualTx.timeout() : 10_000;
+
                     long ts = (txStartTs == null) ? actualTx.startTimestamp().getPhysical() : txStartTs;
 
-                    if (exceptionAllowsImplicitTxRetry(e) && coarseCurrentTimeMillis() - ts < implicitTransactionTimeout) {
+                    if (exceptionAllowsImplicitTxRetry(e) && coarseCurrentTimeMillis() - ts < timeout) {
```

Comment:
How about extracting the `if` condition to a method? It's used a few times in this same form